### DISCUSSION TITLE
Clamp camera XYZ position between 2 values [Evercam]

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#3169] [Q&A]
Hi ITwin team,

We are trying to clamp the camera position whenever we use the camera rotate or pan tool.
For example to be used to prevent the camera on going under ground level.

Is there an even on the camera whenever the camera position changes ?

**Comment by pmconne** [comment]
[Viewport.onViewChanged](https://www.itwinjs.org/reference/core-frontend/views/viewport/onviewchanged/) is invoked immediately every time anything about the viewport changes, including its camera position.
If you were going to try to enforce "camera above ground" in a listener for that event, you'd need  to be very careful to avoid doing expensive computations and/or responding to the onViewChanged callbacks you would trigger.
Maybe @bbastings has thought about the feasibility of making the viewing tools optionally enforce this behavior.

**Comment by bbastings** [comment]
What is the intent behind preventing the user from getting the view they are trying to set? Is the thought that they would do this by accident or something?

**Reply by salah-walid** [reply]
The main reason is to block the user from going under the terrain

**Reply by pmconne** [reply]
Right - Cesium has this behavior by default (they made it optional at our request a long time ago).

**Reply by salah-walid** [reply]
Can we switch it on in ITwin ?

**Comment by bbastings** [comment]
No, it's not currently an option.
